### ISSUE TITLE
Middle Earth is Broken

### ISSUE THREAD

**Issue by RussianSanity** [GreatOrator/TolkienCraftII#114]
The network I'm with is trying to setup a TolkienCraft II server. (Most recent v2.9.4)

We're testing it out before we make it go public to the server since we've never done a server like this before. Our first attempt to get the map on the server failed, as did the second. We downloaded the map from the browser after going to "Download and Install Map" on the launcher. That took us to this link: https://addons-origin.cursecdn.com/files/2314/430/TolkienCraftIIv2.9.x.-.RC3.-.Template.zip
The third attempt, we copied a brand new world from SinglePlayer and threw it on the server. It worked!

So another staff member and myself began playing after all that.. 
Overworld seems fine but we've only tested the first town and quest chapter. 
Middle Earth however... It doesn't spawn you where you're supposed to (in the tented feasting area) like you do in SinglePlayer. 
We went through the ring portal together (probably a stupid idea to begin with) and it took a while before the server timed out. When we logged back on the server, it spawned us on the side of a hill with the chest full of pouches to put things into and the red book. We figured it was a random spawn, so we went searching for the nearest town. Hobbit homes, hobbits, and wildlife were found but no villages. We went to three more towns, same thing. The terrain looks like it's supposed to but there's no villages, not even Hobbiton! 

So I went and grabbed my "MiddleEarth" folder and copied it over to /world/MiddleEarth/ to overwrite the one on the server.
Started it up, nothing happened to the world.
So I deleted the one on the server and put my own again in it's place.
Nothing happened. 

With essentials, I tried /whereami to find out what DIM it was in; thinking maybe it was named something different. 
It said I was in World, as seen here: https://gyazo.com/d3e924ed32037b91866ae7cbd80ed52c (Top "world" is from when I was in Middle Earth and the bottom one is the dimension I'm in when I'm at the starting docks area in the first town.
So I did /dt to see the list of worlds, it says there's THREE loaded worlds named "world"
As seen here: https://gyazo.com/c6a64e04db2d59d08270d49c3da68834
I tried to TP to the other "world"s but it just sent me back to the starting dock area.

Now my questions: how on earth are there three different "world"s?
And how do I go about deleting middle earth so I can import a save from SinglePlayer so that can be back to as it should be? 
tldr: what is the name of the world file I need to delete? 

Should note that we're running Kcauldron version: 
1.7.10-R0.1-SNAPSHOT (MC:1.7.10) (Implementing API version 1.7.10-R0.1-SNAPSHOT)

**Comment by GreatOrator** [comment]
Sounds like you had the world uploaded incorrectly.  There are a total of 4 parts, the main template and 3 content packs.  The main template when extracted will by default be "world" the three content packs you simply take the region folder out of the archive...nothing else...and drop it into the middle earth folder inside world.  the level.dat files in the content packs are there simply because of a limitation with how curse allows world files to be uploaded.

**Comment by GreatOrator** [comment]
the easier way is to simly use the map download in the menu in single player to get all 4 parts.  once download is complete, just upload the entire world folder to your server

**Comment by RussianSanity** [comment]
So we've done as you said, downloaded the template and middle earth region files and thrown them in a folder named "world" and in the folder "MiddleEarth" and it still didn't work.

We uploaded the files separately via. downloading from in the launcher, still didn't generate villages such as Hobbiton. - we did this previously, as I described. 

This is where Hobbiton is and what it looks like on our end to give you an example of what we see: https://gyazo.com/fd1da2f2e77e9c60351ae77fa705939b

It generates everything besides the actual villages built for the areas. 

If you have any advice or solutions.. Please write back. I can give our personal teamspeak or discord if you would like, if that's an easier way to communicate for you. The network I'm with would really like to host this pack.

**Comment by teagan75** [comment]
It is strange that you are having issues. If you tp to Hobbiton via the lotr mod map, you will not find it there. It is closer to Waymeet I think it's called. You should only have 1 save called world. The default map template you downloaded should have Hobbiton in it. The dlc world folders should not be in your instance. The region files from those folders (world/MiddleEarth/regions) should be copied and placed in the same folder but in your instance save.

**Comment by RussianSanity** [comment]
So we've done as you said, downloaded the template and middle earth region files (ONLY) and throw them in their appropriate folder (the regions for Middle Earth I mean) and started the server up... When I go into Middle Earth it still doesn't spawn you where it should. It should spawn you in a tented feasting area, right? That's where I've spawned on two different saves so I'm assuming so.

We also have uploaded the files separately via. downloading from in the launcher, still didn't generate villages such as Hobbiton. - we did this previously, as I described.

This is where Hobbiton is and what it looks like on our end to give you an example of what we see: https://gyazo.com/fd1da2f2e77e9c60351ae77fa705939b

It generates everything besides the actual villages built for the areas. Deer, Hobbits, Hobbit homes, windmills, farms/barns, random ruins on the side of the road, little feasting tables/chairs on the side of the road, and even an inn at Overhill.

I've also walked down to Waymeet and there's still nothing. Nothing in the surrounding areas either.

The only thing I forgot to mention in my last post was that we disabled "PerfectSpawn" mod because it was giving errors and crashing the server. However it doesn't seem like it would impact the game too much, does it?

If you have any advice or solutions.. Please write back. I can give our personal teamspeak or discord if you would like, if that's an easier way to communicate for you. The network I'm with would really like to host this pack.

**Comment by teagan75** [comment]
Could you do me another favor? check to make sure there isn't a folder for dimension 100 in your world save file.

**Comment by GreatOrator** [comment]
What teagan said above and yes, perfect spawn being disabled is a problem as you will not necessarily spawn where necessary in the overworld...If you can, take a screen shot of the folder structure of your world save.

**Comment by RussianSanity** [comment]
(Sorry about deleting my posts earlier. Just frustrated and then went to go and try new things to get it all going. It's been.. One Hell of a time with this.)

So we wiped all the server files out and started new because I realized that my stupidity and lack of sleep made me throw the level.dat files in with the regions for MiddleEarth like you told me not to.
We tried (and it should work if it works just like SinglePlayer) going into middle earth without adding the content packs. Still the same issue. No villages are spawning but everything else is.

Also we can't have PerfectSpawn on because it requires craftbukkit, which we're not running, and would continually restart the server even before loading the mods in.

As for the files, here: 
https://gyazo.com/739718bb70e3acc1e9e485963e7a6244
https://gyazo.com/31344880d49a7087f88e91a2de0b6903
https://gyazo.com/7380cdd60f3df945ca04f4d907565ef5
https://gyazo.com/b2b0f65c81e8a2acdc27e6fac60991ed

Again, this is without the content packs adding the region files to MiddleEarth.

There is no folder for DIM100, but there is for DIM-100, which if my memory serves me right is the Deep Dark.

**Comment by GreatOrator** [comment]
So, perfect spawn should not require any plugin managers since it is a forge mod...so that is weird your system is having issues with it.  As for the middle earth folder, I think I know what your problem is.

Let me first start out by saying that I do not typically provide support for anything other than forge based servers, however, I have learned that in many cases, plugin based servers (aka bukkit) can not use named dimension folders.

That said, try renaming MiddleEarth to DIM100 and then start up your server, Utumno is DIM101 for the record.

**Comment by RussianSanity** [comment]
We greatly appreciate your help through all of this. It's been very helpful and knowledgeable (at least to me)
Now let me say this: YOU ARE AMAZING. Such a simple fix! Thank you so much! You don't know how many hours I've personally worked on this and wracked my brain over this.

**Comment by GreatOrator** [comment]
Glad to help, and you can thank other server admins I have worked with that allowed me to gain such knowledge :D
